javagen/src/main/java/com/azure/autorest/util/returntype/ReturnTypeDescriptionAssembler.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+
+package com.azure.autorest.util.returntype;
+
+import com.azure.autorest.extension.base.plugin.NewPlugin;
+import com.azure.autorest.extension.base.plugin.PluginLogger;
+import com.azure.autorest.model.clientmodel.ClassType;
+import com.azure.autorest.model.clientmodel.GenericType;
+import com.azure.autorest.model.clientmodel.IType;
+import com.azure.core.http.rest.Response;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReturnTypeDescriptionAssembler {
+    private final PluginLogger logger;
+    private final ReturnTypeDescriptionHandlerRegistry handlerRegistry;
+
+    public ReturnTypeDescriptionAssembler(NewPlugin host) {
+        logger = new PluginLogger(host, ReturnTypeDescriptionAssembler.class);
+        handlerRegistry = new ReturnTypeDescriptionHandlerRegistry();
+        handlerRegistry.addFirst(new MonoDescriptionHandler());
```

Comment:
Do we need the complexity of a registry to handle a couple of return types? Can this be simplified in the ClientMethodTemplate to add the description based on return type?